pkg/controller/atlasproject/alert_configurations.go

```diff
@@ -317,3 +337,35 @@ func isAlertConfigSpecEqualToAtlas(logger *zap.SugaredLogger, alertConfigSpec md
 
 	return true
 }
+
+func canAlertConfigurationReconcile(ctx context.Context, atlasClient mongodbatlas.Client, protected bool, project *mdbv1.AtlasProject, logger *zap.SugaredLogger) (bool, error) {
+	if !protected {
+		return true, nil
+	}
+
+	latestConfig := &mdbv1.AtlasProjectSpec{}
+	latestConfigString, ok := project.Annotations[customresource.AnnotationLastAppliedConfiguration]
+	if ok {
+		if err := json.Unmarshal([]byte(latestConfigString), latestConfig); err != nil {
+			return false, err
+		}
+	}
+
+	list, _, err := atlasClient.AlertConfigurations.List(ctx, project.ID(), &mongodbatlas.ListOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	if len(list) == 0 {
+		return true, nil
+	}
+
+	diff := sortAlertConfigs(logger, latestConfig.AlertConfigurations, list)
+	if len(diff.Create) == 0 && len(diff.Delete) == 0 {
+		return true, nil
+	}
+
+	diff = sortAlertConfigs(logger, project.Spec.AlertConfigurations, list)
+
+	return len(diff.Create) == 0 || len(diff.Delete) == 0, nil
```

Comment:
nitpick: these sub-comparisons might self document a bit better if they were more to their own function with a name that explains what we are checking for on each case.

Reply:
This function could be broken into 3 or 4 pieces following the same argument, I believe right now it still has a single responsibility, and doesn't make sense to have micro functions being reused by it.